backend/src/main/java/com/soen341/instagram/service/impl/CommentService.java

```diff
@@ -0,0 +1,161 @@
+package com.soen341.instagram.service.impl;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Service;
+
+import com.soen341.instagram.dao.impl.AccountRepository;
+import com.soen341.instagram.dao.impl.CommentRepository;
+import com.soen341.instagram.dao.impl.PictureRepository;
+import com.soen341.instagram.exception.comment.CommentLengthTooLongException;
+import com.soen341.instagram.exception.comment.CommentNotFoundException;
+import com.soen341.instagram.exception.like.MultipleLikeException;
+import com.soen341.instagram.exception.picture.InvalidIdException;
+import com.soen341.instagram.exception.picture.PictureNotFoundException;
+import com.soen341.instagram.model.Account;
+import com.soen341.instagram.model.Comment;
+import com.soen341.instagram.model.Picture;
+
+@Service("commentService")
+public class CommentService
+{
+	@Autowired
+	private CommentRepository commentRepository;
+	@Autowired
+	private AccountRepository accountRepository;
+	@Autowired
+	private PictureRepository pictureRepository;
+	@Autowired
+	@Qualifier("UserDetailsService")
+	private UserDetailsService userDetailsService;
+
+	private static int maxCommentLength = 250;
+
+	public Comment createComment(final String commentContent, final long pictureId)
+	{
+		if (commentContent.length() > maxCommentLength)
+		{
+			throw new CommentLengthTooLongException("Comment length exceeds " + maxCommentLength + " characters");
+		}
+
+		final Account account = getCurrentUser();
+		final Optional<Picture> picture = pictureRepository.findById(pictureId);
+
+		if (!picture.isPresent())
+		{
+			throw new PictureNotFoundException();
+		}
+
+		final Comment comment = new Comment();
+		comment.setComment(commentContent);
+		comment.setCreated(new Date());
+		comment.setAccount(account);
+		comment.setPicture(picture.get());
+
+		commentRepository.save(comment);
+		return comment;
+	}
+
+	public int likeComment(final long commentId)
+	{
+		final Comment comment = findComment(commentId);
+		final Set<Account> likedBy = comment.getLikedBy();
+		final boolean addedSuccessfully = likedBy.add(getCurrentUser());
+		if (!addedSuccessfully)
+		{
```

Comment:
Is the only reason this can fail is if you have already liked it? Just in terms of clarity, I think the variable name addedSuccessfully can be changed to reflect what you are actually checking for (multiple likes).

Reply:
Yes exactly, and totally agree on changing the name of the variable to something like `multipleLikes` . Noted for next sprint.